lib/step.rb

```diff
@@ -114,17 +114,24 @@ def puts_full_output
     end
 
     def annotation_location(name)
-      formula = Formulary.factory(name)
+      formula, formula_path = begin
+        f = Formulary.factory(name)
+        [f, f.path]
+      rescue FormulaUnavailableError
+        [nil, @repository.glob("**/#{name}*").first]
+      end
+
       method_sym = command.second.to_sym
-      method_location = formula.method(method_sym).source_location if formula.respond_to?(method_sym)
 
-      if method_location.present? && (method_location.first == formula.path.to_s)
-        method_location
-      else
-        [formula.path, nil]
+      formula_line = if output? && (line = output.lines.find { |l| l.start_with? formula_path })
```

Comment:
```suggestion
      formula_line = if output? && (line = output.lines.grep(/^#{formula_path}/)
```

Reply:
This could return multiple lines, though, and I think I only want to find the first one.

Reply:
Gotcha, no worries then.